compiler/damlc/daml-stdlib-src/DA/Next/Map.daml

```diff
@@ -55,17 +55,30 @@ instance MapKey Text where
   keyToText x = x
   keyFromText x = x
 
+keyFromTextNote : Text -> Text -> Text
+keyFromTextNote target t =
+  "keyFromText: could not parse " <> show t <> " as '" <> target <> "'"
+
 instance MapKey Party where
   keyToText = partyToText
-  keyFromText = fromSome . partyFromText
+  keyFromText t =
+    fromSomeNote
+      (keyFromTextNote "Party" t)
+      (partyFromText t)
```

Comment:
Welcome to another episode of "Why strict languages are tricky (at least for the Haskell minded)":

I'm a bit worried about unconditionally computing the error message even if we don't need it. Some functions like `toList` call `keyFromText` on all keys of a map, which I'm tempted to call a "hot loop". Since we only need to change three instances of `keyFromText`, I'm inclined to manually inline `fromSomeNote` everywhere, like so: 
```suggestion
    case partyFromText t of
      None -> keyFromTextNote "Party" t
      Some p -> p
```

Reply:
fixed

Reply:
Thank you.